dotnet publish error NU1605 (version 8.2.3, framework netstandard2.0, runtime linux-x64)

what am I doing

```
dotnet new classlib
dotnet add package JWT
dotnet publish -f netstandard2.0 -r linux-x64
```

what happens

* `dotnet publish` fails with NU1605

what do I expect

* `dotnet publish` succeeds

csproj

```
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <TargetFramework>netstandard2.0</TargetFramework>
  </PropertyGroup>

  <ItemGroup>
    <PackageReference Include="JWT" Version="8.2.3" />
  </ItemGroup>

</Project>
```

workaround

```
dotnet add package Newtonsoft.Json -v 10.0.1
```

logs

```
[brandon@localhost foo]$ dotnet new classlib
The template "Class library" was created successfully.

Processing post-creation actions...
Running 'dotnet restore' on /home/brandon/src/scratch/foo/foo.csproj...
  Restore completed in 82.77 ms for /home/brandon/src/scratch/foo/foo.csproj.

Restore succeeded.

[brandon@localhost foo]$ dotnet add package JWT
  Writing /tmp/tmp5po6bT.tmp
info : Adding PackageReference for package 'JWT' into project '/home/brandon/src/scratch/foo/foo.csproj'.
info : Restoring packages for /home/brandon/src/scratch/foo/foo.csproj...
info :   GET https://api.nuget.org/v3-flatcontainer/jwt/index.json
info :   OK https://api.nuget.org/v3-flatcontainer/jwt/index.json 374ms
info : Package 'JWT' is compatible with all the specified frameworks in project '/home/brandon/src/scratch/foo/foo.csproj'.
info : PackageReference for package 'JWT' version '8.2.3' added to file '/home/brandon/src/scratch/foo/foo.csproj'.
info : Committing restore...
info : Generating MSBuild file /home/brandon/src/scratch/foo/obj/foo.csproj.nuget.g.props.
info : Writing assets file to disk. Path: /home/brandon/src/scratch/foo/obj/project.assets.json
log  : Restore completed in 638.02 ms for /home/brandon/src/scratch/foo/foo.csproj.
[brandon@localhost foo]$ dotnet publish -f netstandard2.0 -r linux-x64
Microsoft (R) Build Engine version 16.4.0+e901037fe for .NET Core
Copyright (C) Microsoft Corporation. All rights reserved.

/home/brandon/src/scratch/foo/foo.csproj : error NU1605: Detected package downgrade: System.IO.FileSystem.Primitives from 4.3.0 to 4.0.1. Reference the package directly from the project to select a different version. 
/home/brandon/src/scratch/foo/foo.csproj : error NU1605:  foo -> JWT 8.2.3 -> Newtonsoft.Json 9.0.1 -> System.Xml.ReaderWriter 4.0.11 -> System.IO.FileSystem 4.0.1 -> runtime.unix.System.IO.FileSystem 4.3.0 -> System.IO.FileSystem.Primitives (>= 4.3.0) 
/home/brandon/src/scratch/foo/foo.csproj : error NU1605:  foo -> JWT 8.2.3 -> Newtonsoft.Json 9.0.1 -> System.Xml.ReaderWriter 4.0.11 -> System.IO.FileSystem.Primitives (>= 4.0.1)
/home/brandon/src/scratch/foo/foo.csproj : error NU1605: Detected package downgrade: System.Text.Encoding.Extensions from 4.3.0 to 4.0.11. Reference the package directly from the project to select a different version. 
/home/brandon/src/scratch/foo/foo.csproj : error NU1605:  foo -> JWT 8.2.3 -> Newtonsoft.Json 9.0.1 -> System.Xml.ReaderWriter 4.0.11 -> System.IO.FileSystem 4.0.1 -> runtime.unix.System.IO.FileSystem 4.3.0 -> System.Text.Encoding.Extensions (>= 4.3.0) 
/home/brandon/src/scratch/foo/foo.csproj : error NU1605:  foo -> JWT 8.2.3 -> Newtonsoft.Json 9.0.1 -> System.Text.Encoding.Extensions (>= 4.0.11)
  Restore failed in 200.42 ms for /home/brandon/src/scratch/foo/foo.csproj.
[brandon@localhost foo]$ dotnet add package Newtonsoft.Json -v 10.0.1
  Writing /tmp/tmpNYVe34.tmp
info : Adding PackageReference for package 'Newtonsoft.Json' into project '/home/brandon/src/scratch/foo/foo.csproj'.
info : Restoring packages for /home/brandon/src/scratch/foo/foo.csproj...
info : Package 'Newtonsoft.Json' is compatible with all the specified frameworks in project '/home/brandon/src/scratch/foo/foo.csproj'.
info : PackageReference for package 'Newtonsoft.Json' version '10.0.1' added to file '/home/brandon/src/scratch/foo/foo.csproj'.
info : Committing restore...
info : Generating MSBuild file /home/brandon/src/scratch/foo/obj/foo.csproj.nuget.g.props.
info : Writing assets file to disk. Path: /home/brandon/src/scratch/foo/obj/project.assets.json
log  : Restore completed in 162.83 ms for /home/brandon/src/scratch/foo/foo.csproj.
[brandon@localhost foo]$ dotnet publish -f netstandard2.0 -r linux-x64
Microsoft (R) Build Engine version 16.4.0+e901037fe for .NET Core
Copyright (C) Microsoft Corporation. All rights reserved.

  Restore completed in 196.25 ms for /home/brandon/src/scratch/foo/foo.csproj.
  foo -> /home/brandon/src/scratch/foo/bin/Debug/netstandard2.0/linux-x64/foo.dll
  foo -> /home/brandon/src/scratch/foo/bin/Debug/netstandard2.0/linux-x64/publish/
```

--- a/src/JWT.Extensions.AspNetCore/JWT.Extensions.AspNetCore.csproj
+++ b/src/JWT.Extensions.AspNetCore/JWT.Extensions.AspNetCore.csproj
@@ -10,7 +10,7 @@
     <Authors>Alexander Batishchev</Authors>
     <PackageTags>jwt;json;asp.net;asp.net core;.net core</PackageTags>
     <PackageLicense>MIT</PackageLicense>
-    <Version>7.1.0-beta2</Version>
+    <Version>7.2.0</Version>
     <FileVersion>7.0.0.0</FileVersion>
     <AssemblyVersion>7.0.0.0</AssemblyVersion>
     <RootNamespace>JWT</RootNamespace>
@@ -31,7 +31,7 @@
   <ItemGroup>
     <PackageReference Include="Microsoft.AspNetCore.Authentication" Version="2.2.0" />
     <PackageReference Include="Microsoft.AspNetCore.Http.Abstractions" Version="2.2.0" />
-    <PackageReference Include="Newtonsoft.Json" Version="9.0.1" />
+    <PackageReference Include="Newtonsoft.Json" Version="10.0.3" />
   </ItemGroup>
 
   <ItemGroup>

--- a/src/JWT.Extensions.Owin/JWT.Extensions.Owin.csproj
+++ b/src/JWT.Extensions.Owin/JWT.Extensions.Owin.csproj
@@ -10,7 +10,7 @@
     <Authors>Alexander Batishchev</Authors>
     <PackageTags>jwt;json;asp.net;asp.net core;.net core</PackageTags>
     <PackageLicense>MIT</PackageLicense>
-    <Version>6.0.0-alpha2</Version>
+    <Version>6.1.0</Version>
     <FileVersion>6.0.0.0</FileVersion>
     <AssemblyVersion>6.0.0.0</AssemblyVersion>
     <RootNamespace>JWT</RootNamespace>
@@ -30,7 +30,7 @@
 
   <ItemGroup>
     <PackageReference Include="Microsoft.Owin" Version="3.0.1" />
-    <PackageReference Include="Newtonsoft.Json" Version="9.0.1" />
+    <PackageReference Include="Newtonsoft.Json" Version="10.0.3" />
   </ItemGroup>
 
   <ItemGroup>

--- a/src/JWT/JWT.csproj
+++ b/src/JWT/JWT.csproj
@@ -21,7 +21,7 @@
     <Authors>Alexander Batishchev, John Sheehan, Michael Lehenbauer</Authors>
     <PackageTags>jwt;json</PackageTags>
     <PackageLicense>CC0-1.0</PackageLicense>
-    <Version>8.2.3</Version>
+    <Version>8.4.0-beta1</Version>
     <FileVersion>8.0.0.0</FileVersion>
     <AssemblyVersion>8.0.0.0</AssemblyVersion>
     <RootNamespace>JWT</RootNamespace>
@@ -41,7 +41,7 @@
 
   <ItemGroup>
     <PackageReference Include="Microsoft.SourceLink.GitHub" Version="1.0.0" PrivateAssets="All" />
-    <PackageReference Include="Newtonsoft.Json" Version="9.0.1" />
+    <PackageReference Include="Newtonsoft.Json" Version="10.0.1" />
   </ItemGroup>
 
   <ItemGroup Condition="'$(TargetFramework)' == 'netstandard1.3' OR '$(TargetFramework)' == 'netstandard2.0' OR '$(TargetFramework)' == 'net50'">

--- a/src/JWT/JWT.csproj
+++ b/src/JWT/JWT.csproj
@@ -41,7 +41,7 @@
 
   <ItemGroup>
     <PackageReference Include="Microsoft.SourceLink.GitHub" Version="1.0.0" PrivateAssets="All" />
-    <PackageReference Include="Newtonsoft.Json" Version="10.0.1" />
+    <PackageReference Include="Newtonsoft.Json" Version="10.0.3" />
   </ItemGroup>
 
   <ItemGroup Condition="'$(TargetFramework)' == 'netstandard1.3' OR '$(TargetFramework)' == 'netstandard2.0' OR '$(TargetFramework)' == 'net50'">

